client/src/components/Applications/AppLayout/AppLayout.js

```diff
@@ -1,24 +1,31 @@
 import React from 'react';
 import { Route, useRouteMatch } from 'react-router-dom';
 import { useDispatch, useSelector, shallowEqual } from 'react-redux';
-import { LoadingSpinner, Section } from '_common';
+import { LoadingSpinner, Section, SectionMessage } from '_common';
 import './AppLayout.global.css';
 import AppBrowser from '../AppBrowser/AppBrowser';
 import { AppDetail, AppPlaceholder } from '../AppForm/AppForm';
 
 const AppsLayout = () => {
   const { params } = useRouteMatch();
-  const { loading, categoryDict } = useSelector(
+  const { loading, categoryDict, error } = useSelector(
     state => ({
       loading: state.apps.loading,
+      error: state.apps.error,
       categoryDict: state.apps.categoryDict
     }),
     shallowEqual
   );
-
+  if (error.isError) {
+    return (
+      <div id="appDetail-wrapper" className="has-message  appDetail-error">
+        <SectionMessage type="warning">Something went wrong.</SectionMessage>
+      </div>
+    );
+  }
   return (
     <>
-      {loading && !Object.keys(categoryDict).length ? (
```

Comment:
Was this removed for a particular reason?

Reply:
@rstijerina Yes, that line was unnecessary, so I removed it .

Reply:
@rstijerina , we checked the reducer and confirmed that loading was always set correctly

Reply:
I see. @nathanfranklin I think I may have added that to make the LoadingSpinner only show if `loading` is `true` AND if there are no apps in the `categoryDict`, to allow for a seamless reloading of apps. Thoughts?

Reply:
Gotcha.  That makes sense. 🎗️  i'll take a look again tomorrow or thursday at this PR again but I think i follow. 👍